Assets/Scripts/Jogador/CharacterSelector.cs

```diff
@@ -0,0 +1,40 @@
+﻿using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace Jogador
+{
+	public class CharacterSelector : IPlayerSelection
+	{
+		[SerializeField] private GameObject confirmedPlayers;
+		[SerializeField] private SingleCharacterSelector[] singleCharacterSelector;
+
+		public override void ShowElements()
+		{
+			confirmedPlayers.SetActive(true);
+		}
+
+		protected override void HideComponent()
+		{
+			confirmedPlayers.SetActive(false);
+		}
+
+		public override IPlayerDetails[] GetPlayers()
+		{
+			var players = new IPlayerDetails[5];

```

Comment:
Em vez de ir adicionando em um array cria uma List<IplayerDetails>() e em vez de atribuir players[i] = getCharacter(), vc pode ver se foi escolhido o jogador tipo
`if(singleCharacterSelector[i].isReady())`
e só se o jogador estivesse pronto usar o `players.add(singleCharacterSelector.GetPlayerCharacter())`

Pq disso? Pq vai ter um array com muitos buracos vazios, depois na hora do turno para tratar fica meio escondido que eu to devolvendo um array com buracos, é algo n muito legal que pode dar uns null pointers.

Dai no caso se tiver 2-3 jogadores devolve um array com tamanho 2-3 e não um array com tamanho 5.

Dai nessa lógica na linha 30 iria retornar players.toArray();